app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -85,6 +104,6 @@ def create_post
   end
 
   def post_params
-    params.require(:discussion_post).permit(:title, :text, :parent_id)
```

Comment:
No need to use `parent_id` anymore, because the child post can be built from a parent post.